clang 12 errors (linux)

When attempting to build with clang 12, I get this:

[  0%] Building CXX object Code/Framework/AzCore/CMakeFiles/AzCore.dir/AzCore/Component/EntityUtils.cpp.o
cd /home/spot/rpmbuild/BUILD/o3de/x86_64-redhat-linux-gnu/Code/Framework/AzCore && /usr/bin/clang++ -DAZ_BUILD_CONFIGURATION_TYPE=\"release\" -DLINUX -DLINUX64 -DNDEBUG -DRELEASE -D_HAS_EXCEPTIONS=0 -D_RELEASE -D__linux__ -I/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/. -I/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/Platform/Linux -I/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/Platform/Common -isystem /usr/include/lua-5.1 -isystem /home/spot/rpmbuild/BUILD/o3de/3rdParty/rapidxml/1.13-modified.1/include -fno-exceptions -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Werror -Wrange-loop-analysis -Wno-unknown-warning-option -Wno-#pragma-messages -Wno-absolute-value -Wno-defaulted-function-deleted -Wno-dynamic-class-memaccess -Wno-format-security -Wno-inconsistent-missing-override -Wno-invalid-offsetof -Wno-multichar -Wno-parentheses -Wno-reorder -Wno-self-assign -Wno-switch -Wno-tautological-compare -Wno-undefined-var-template -Wno-unknown-pragmas -Wno-unused-function -Wno-unused-private-field -Wno-unused-value -Wno-unused-variable -Wno-non-pod-varargs -Wno-unused-lambda-capture -Wno-nonportable-include-path -fPIC -msse4.1  -O2 -std=c++17 -o CMakeFiles/AzCore.dir/AzCore/Component/EntityUtils.cpp.o -c /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/AzCore/Component/EntityUtils.cpp
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/AzCore/UnitTest/MockComponentApplication.cpp:13:
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/UnitTest/MockComponentApplication.h:15:
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/Component/ComponentApplicationBus.h:15:
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/EBus/EBus.h:25:
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/EBus/BusImpl.h:25:
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/EBus/Internal/BusContainer.h:14:
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/std/functional.h:24:
In file included from /home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/std/allocator.h:17:
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:165:35: error: use of template template parameter 'T' requires template arguments
            static_assert(false_v<T>, "Missing specialization for this template. Make sure it's registered for type info support.");
                                  ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:162:46: note: template is declared here
        template<template<typename...> class T>
                 ~~~~~~~~~~~~~~~~~~~~~       ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:173:35: error: use of template template parameter 'T' requires template arguments
            static_assert(false_v<T>, "Missing specialization for this template. Make sure it's registered for type info support.");
                                  ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:170:51: note: template is declared here
        template<template<AZStd::size_t...> class T>
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~       ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:181:35: error: use of template template parameter 'T' requires template arguments
            static_assert(false_v<T>, "Missing specialization for this template. Make sure it's registered for type info support.");
                                  ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:178:58: note: template is declared here
        template<template<typename, AZStd::size_t> class T>
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~       ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:189:35: error: use of template template parameter 'T' requires template arguments
            static_assert(false_v<T>, "Missing specialization for this template. Make sure it's registered for type info support.");
                                  ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:186:68: note: template is declared here
        template<template<typename, typename, AZStd::size_t> class T>
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~       ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:197:35: error: use of template template parameter 'T' requires template arguments
            static_assert(false_v<T>, "Missing specialization for this template. Make sure it's registered for type info support.");
                                  ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:194:78: note: template is declared here
        template<template<typename, typename, typename, AZStd::size_t> class T>
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~       ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:205:35: error: use of template template parameter 'T' requires template arguments
            static_assert(false_v<T>, "Missing specialization for this template. Make sure it's registered for type info support.");
                                  ^
/home/spot/rpmbuild/BUILD/o3de/Code/Framework/AzCore/./AzCore/RTTI/TypeInfo.h:202:68: note: template is declared here
        template<template<typename, AZStd::size_t, typename> class T>
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~       ^

--- a/Code/CryEngine/CryCommon/ISystem.h
+++ b/Code/CryEngine/CryCommon/ISystem.h
@@ -1154,13 +1154,13 @@ struct DiskOperationInfo
         return *this;
     }
 
-    DiskOperationInfo& operator - (const DiskOperationInfo& rv)
+    DiskOperationInfo operator - (const DiskOperationInfo& rv)
     {
         DiskOperationInfo res(*this);
         return res -= rv;
     }
 
-    DiskOperationInfo& operator + (const DiskOperationInfo& rv)
+    DiskOperationInfo operator + (const DiskOperationInfo& rv)
     {
         DiskOperationInfo res(*this);
         return res += rv;

